### PR TITLE
Wings and tails give anyone better tackling values. Moths and lizards that lose theirs will have worse values instead.

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
Late and atomized version of #67791 with some extra changes.

Changes are:

1 - Anyone with wings will have a +2 tackle offense (same as riot armor or gigantism mutation)
2 - Moths without or burnt wings will have -2 tackle offense (same as dwarfism or clumsy)
3 - Anyone with a lizard tail that is wagging it, will have 1+ tackle defense (same as being fat)
4 - Lizards without a tail will have -1 tackle defense (same as being slightly drunk)
5 - Fly people have a higher chance of breaking their spine after splatting themselves into a window/wall (+6 same as clumsy)

Made 3 new traits and gave them to Lizards, Moths and Flypeople, swapped the old `islizard` to check for one of these traits instead.

This way it is easier for anyone that reads these species' code to know that they have an unique interaction with our tackling system without having to dig into component code.
And thanks for the suggestion Kapu.

## Why It's Good For The Game
This makes Moths have an interesting round start niche as security and push them to behave differently than humans if they decide to opt into tackling.

It is fun to steal flashbangs and exploit the weakness that Moth officers have right now, if moths are better at tackling and more likely to use it, it opens another thing you can exploit when engaging against security.

The breakpoints for tackling are on windows of 2 so while it might look like a lot (Riot suit value), if I gave only 1 for moths they would have a 50% chance of being better than other species while tackling.
Inconsistencies are not interesting on species. It would be quite annoying if plasmamen had a coin toss on being cold resistant or self igniting every time they engage with the system.

The flypeople change is there just because the idea of a Fly officer smacking themselves into a window and breaking their spine is funny.

And as was requested by maintainers, it now checks for anyone with wings and lizard tails for the tackle buff, go have your fun Frankstein builders and Officers that can get Xeno to give you a wing potion.

## Changelog

:cl: Guillaume Prata
balance: Anyone with wings or a wagging lizard tail will have a small bonus to their tacklling offense/defense.
balance: Moths and lizards that lose their wings/tail will have worse tackling values instead.
balance: Fly people don't take splatting themselves on windows/walls easily and have a higher chance to get a bad result from that.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Did you get maintainer approval for your species change?

**Comment by GuillaumePrata** [comment]
> Did you get maintainer approval for your species change?

Ryll gave me approval on the original PR, I assume I didn't need to ask again even if I took an long time to remake this.

**Comment by Mothblocks** [comment]
Okay just making sure

**Comment by wesoda25** [comment]
Why not check for wings as opposed to species? (I want to make my security apex predator with asimov protection)

**Comment by AlamoTurtle** [comment]
Moths ought to just be able to fly further with tackles regardless of the gloves worn, as extra "flight".

**Comment by GuillaumePrata** [comment]
> fly further

While this could be flavorful, it would make Moths be a lot worse than normal security at tackling.
You rarely miss tackles because you didn't go far enough. It would make you fly into a wall every time you miss which is a death sentences mid combat.

File: code/datums/components/tackle.dm [hunk 1]
```diff
@@ -309,6 +309,13 @@
 	if(HAS_TRAIT(sacker, TRAIT_GIANT))
 		attack_mod += 2
 
+	if(ismoth(sacker))
+		var/obj/item/organ/external/wings/moth/sacker_wing = sacker.getorganslot(ORGAN_SLOT_EXTERNAL_WINGS)
+		if(!sacker_wing || sacker_wing.burnt) // moths without wing/burnt ones can't tackle properly
+			attack_mod -= 2
+		else
+			attack_mod += 2 // healty wings are used as extra propulsion
+
```
Comment (Kapu1178) [inline code/datums/components/tackle.dm:318]: Hardcoding species checks is bad. Do not do it

```suggestion
if(locate(/obj/item/organ/external/wings/moth) in sacker.dna?.species.external_organs)
  var/obj/item/organ/external/wings/moth/sacker_wing = sacker.getorganslot(ORGAN_SLOT_EXTERNAL_WINGS)
  if(!sacker_wing || sacker_wing.burnt) // moths without wing/burnt ones can't tackle properly
	  attack_mod -= 2
  else
	  attack_mod += 2 // healty wings are used as extra propulsion

```

File: code/datums/components/tackle.dm [hunk 2]
```diff
@@ -373,6 +380,9 @@
 	if(HAS_TRAIT(user, TRAIT_CLUMSY))
 		oopsie_mod += 6 //honk!
 
+	if(isflyperson(user))
+		oopsie_mod += 6 // flies don't take smacking into a window/wall easily
```
Comment (Kapu1178) [inline code/datums/components/tackle.dm:384]: Don't hardcode species checks. Make this a trait or something.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/datums/components/tackle.dm (hunk 2 above)
Comment (itseasytosee) [reply code/datums/components/tackle.dm:384]: Better yet, make it a var attached to physiology

File: code/datums/components/tackle.dm (hunk 1 above)
Comment (GuillaumePrata) [reply code/datums/components/tackle.dm:318]: I will make it into traits for each species.
It is easier for someone checking a species code to know they are better at tackling without having to dig into the component code.

File: code/datums/components/tackle.dm [hunk 3]
```diff
@@ -309,6 +309,14 @@
 	if(HAS_TRAIT(sacker, TRAIT_GIANT))
 		attack_mod += 2
 
+	if(HAS_TRAIT(sacker, TRAIT_TACKLING_WINGED_ATTACKER))
+		var/obj/item/organ/external/wings/moth/sacker_moth_wing
```
Comment (ShizCalev) [inline code/datums/components/tackle.dm:313]: you forgot something.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Pull request merged**